src/core-sdk-tasks/DownloadFile.cs

```diff
@@ -44,26 +50,47 @@ public override bool Execute()
             {
                 Log.LogMessage(MessageImportance.High, $"Downloading '{Uri}' to '{DestinationPath}'");
 
-                using (var httpClient = new HttpClient())
+                if (!DownloadFromUri(Uri, DestinationPath))
                 {
-                    var getTask = httpClient.GetStreamAsync(Uri);
-
-                    try
+                    if (!string.IsNullOrWhiteSpace(PrivateUri))
                     {
-                        using (var outStream = File.Create(DestinationPath))
+                        Log.LogMessage(MessageImportance.High, $"Couldn't download file '{Uri}' to '{DestinationPath}'. Trying to download file from '{PrivateUri}' instead.");
+
+                        if (!DownloadFromUri(PrivateUri, DestinationPath))
                         {
-                            getTask.Result.CopyTo(outStream);
+                            Log.LogError($"Couldn't download file '{PrivateUri}' to '{DestinationPath}' either.");
+                            return false;
                         }
                     }
-                    catch (Exception)
+                    else
                     {
-                        File.Delete(DestinationPath);
-                        throw;
+                        Log.LogError($"Couldn't download file '{Uri}' to '{DestinationPath}'");
+                        return false;
                     }
                 }
             }
 
             return true;
         }
+        
+        private bool DownloadFromUri(string source, string target)
+        {
+            try
+            {
+                using (var httpClient = new HttpClient())
+                using (var outStream = File.Create(target))
+                {
+                    var getTask = httpClient.GetStreamAsync(source);
+                    getTask.Result.CopyTo(outStream);
+                }
+
+                return true;
+            }
+            catch (Exception)
+            {
+                File.Delete(target);
+                return false;
```

Comment:
Is this meant to swallow all exceptions silently? It looks like all the diagnosis you get is `Couldn't download file` and the urls, which doesn't sound good to me.